src/Visualization/Python/ApplyPointwise.py

```diff
@@ -0,0 +1,313 @@
+                nargs=-1,
+                type=click.Path(exists=True,
+                                file_okay=True,
+                                dir_okay=False,
+                                readable=True))
+@click.option("--subfile-name",
+              "-d",
+              help="Name of volume data subfile within each h5 file.")
+@click.option("--kernel",
+              "-k",
+              "kernels",
+              multiple=True,
+              required=True,
+              help=("Python function(s) to apply to the volume data. "
+                    "Specify as 'path.to.module.function_name', where the "
+                    "module must be available to import. "
+                    "Alternatively, specify just 'function_name' if the "
+                    "function is defined in one of the '--exec' / '-e' "
+                    "files."))
+@click.option("--exec",
+              "-e",
+              "exec_files",
+              type=click.File("r"),
+              multiple=True,
+              help=("Python file(s) to execute before loading kernels. "
+                    "Load kernels from these files with the '--kernel' / '-k' "
+                    "option."))
+@click.option("--input-name",
+              "-i",
+              "map_input_names",
+              multiple=True,
+              callback=parse_input_names,
+              help=("Map of function argument names to dataset names "
+                    "in the volume data file. Specify key-value pairs "
+                    "like 'spatial_metric=SpatialMetric'. If unspecified, "
+                    "the argument name is transformed to CamelCase."))
+def apply_pointwise_command(h5files, subfile_name, kernels, exec_files,
+                            map_input_names, **kwargs):
+    """Apply pointwise functions to volume data
+
+    Run pointwise functions (kernels) over all volume data in the 'H5FILES' and
+    write the output data back into the same files. You can use any Python
+    function as kernel that takes tensors as input arguments and returns a
+    tensor (from 'spectre.DataStructures.Tensor'). The function must be
+    annotated with tensor types, like this:
+
+        def shift_magnitude(
+                shift: tnsr.I[DataVector, 3],
+                spatial_metric: tnsr.ii[DataVector, 3]) -> Scalar[DataVector]:
+            # ...
+
+    Any pybind11 binding of a C++ function will also work, as long as it takes
+    exclusively tensors as arguments. The kernels can be loaded from any
+    available Python module, such as 'spectre.PointwiseFunctions'. You can also
+    execute a Python file that defines kernels with the '--exec' / '-e' option.
+
+    By default, the data for the input arguments are read from datasets in the
+    volume files with the same names, transformed to CamelCase. For example, the
+    input dataset names for the 'shift_magnitude' function above would be
+    'Shift(_x,_y,_z)' and 'SpatialMetric(_xx,_yy,_zz,_xy,_xz,_yz)'. The output
```

Comment:
Maybe add a sentence "That is, the code uses the name `shift` from the function argument, changes it to CamelCase, then reads the `Shift(_x,_y,_z)` datasets into a `tnsr.I[DataVector, 3]` before passing it to the function." This would've made it easier for me to understand.